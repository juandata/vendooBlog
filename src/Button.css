.Button {
    width: 100%;
    height: 80px;
    
}

.ButtonInner  {
height: 22px;
font-family: Montserrat;
font-size: 18px;
font-weight: 600;
color: #ffffff;
text-decoration: none solid rgb(255, 255, 255);
text-align: center;
transition: 0.3s;

display: flex;
    width: 100%;
    height: 100%;
    align-items: center;
    justify-content: center;
}

.ButtonInner:hover {
    opacity: 0;
}
.ButtonEffect {
    width: 100%;
    height: 100%;
    background-position: center;
    background-repeat: no-repeat;
    transition: 0.3s;

}
.ButtonEffect:hover {
    background-image: url('./assets/path.svg')!important;
}