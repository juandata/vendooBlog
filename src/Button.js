import React from "react";

import './Button.css';

const Button = props => {
    const style = {
        backgroundImage : `url(${props.Img})`
    }
    return <button style={style} className="Button"><div className="ButtonEffect"><div className="ButtonInner"><div>View More</div></div></div></button>
}

export default Button;
