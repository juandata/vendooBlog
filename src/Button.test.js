import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import buttonBackgroundImg from './assets/buttonBackground.svg';


it('Button should render correctly with no props', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });

  it('Button mouseover should make image effect with given props', () => {
    const component = shallow(
        <Button Img={buttonBackgroundImg} />
    );
    component
      .simulate('mouseenter');
    expect(component).toMatchSnapshot();
    component.unmount();
  });