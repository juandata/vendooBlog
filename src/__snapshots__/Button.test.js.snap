// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button mouseover should make image effect with given props 1`] = `
<button
  className="Button"
  style={
    Object {
      "backgroundImage": "url(buttonBackground.svg)",
    }
  }
>
  <div
    className="ButtonEffect"
  >
    <div
      className="ButtonInner"
    >
      <div>
        View More
      </div>
    </div>
  </div>
</button>
`;

exports[`Button should render correctly with no props 1`] = `
<button
  className="Button"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <div
    className="ButtonEffect"
  >
    <div
      className="ButtonInner"
    >
      <div>
        View More
      </div>
    </div>
  </div>
</button>
`;
